Added basic backend integration to /profile

asadsid95 opened the pull request:
NOTE: Most of the changes are due to Prettier formatting

Main focus is on the client/src/pages/Profile.js (See below for updates)

- I set up a basic fetch api to a path that should respond with the Profile and Projects of the user:
        - I’m not sure how I can use the objects provided in the response to update the information on the page (I acknowledge that it’s dependent on the user ID AKA ‘only showing YOUR profile and projects’)
        - I’m also not sure what would trigger this fetch API call:
            - Example: for POST /login its the submission of email and password that triggers the fetch API… But for something thats dependent on navigating to the page, I’m not sure how to go about that

asadsid95 committed:
Added basic backend integration to /profile

NOTE: Most of the changes are due to Prettier formatting

Main focus is on the client/src/pages/Profile.js (See below for updates)

- I set up a basic fetch api to a path that should respond with the Profile and Projects of the user:
        - I’m not sure how I can use the objects provided in the response to update the information on the page (I acknowledge that it’s dependent on the user ID AKA ‘only showing YOUR profile and projects’)
        - I’m also not sure what would trigger this fetch API call:
            - Example: for POST /login its the submission of email and password that triggers the fetch API… But for something thats dependent on navigating to the page, I’m not sure how to go about that

aeciorc commented on client/src/pages/Profile.js:
```diff
@@ -1,147 +1,170 @@
-] 
+];
+
+// Purposely didn't put the following as a part of a function's body
```
You would put that inside a `useEffect` hook inside  profilePage, so that it runs when the component finishes loading (https://reactjs.org/docs/hooks-effect.html, https://medium.com/@felippenardi/how-to-do-componentdidmount-with-react-hooks-553ba39d1571). Then, once you have the response, you save in the state using a `useState` hook (https://reactjs.org/docs/hooks-state.html)

asadsid95 committed:
Prettier-formatted client & server folder

asadsid95 committed:
Merge branch 'dev' into Integrating-Profile

asadsid95 committed:
Added fetch GET to /profile/:id

Files of interest:
- client/src/pages/Profile.js
- client/src/pages/Fields.js

fetch GET for /profile/:id (URL path is of interest); followed by storing profile and projects using useState

Passed expertise(array) from profile to fields component but in /Fields.js, it isn't able to iterate over the array and show individual expertise as seperate Chips

        - I expect same problem to happen in /Project.js... It's currently passing a hard-coded array of projects which iterate over to show as a list.

aeciorc commented on client/src/pages/Profile.js:
```diff
@@ -71,57 +71,63 @@ const projectData = [
+
+  useEffect(() => {
+    //fetch(`/profile/${props.match.params.id}`)
+    fetch(`${props.location.pathname}`) // ** URL is equivalent to /profile/:id
```
This is a little obscure, it relies on the backend url being the same as the current path.

aeciorc commented on client/src/pages/Profile.js:
```diff
@@ -71,57 +71,63 @@ const projectData = [
-                src={JHAvatar}
-                className={classes.bigAvatar}
-              />
+              <Avatar alt="James Hampton" src={JHAvatar} className={classes.bigAvatar} />
```
Change the hardcoded name here as well

aeciorc replied:
And make it so that you use the image url from the backend (I think that's been implemented already right?)

aeciorc commented on client/src/pages/Profile.js:
```diff
@@ -137,16 +143,10 @@ export default function ProfilePage() {
-
-          <ProjectList projectData={projectData} />
-
+          <ProjectList projectData={projects} /> {/*the prop would be changed with projects state */}
```
Is it working rendering the projects as expected?

asadsid95 replied:
Currently, no projects are a part of a profile so that's why I wasnt able to test that out

asadsid95 committed:
Merge branch 'dev' into Integrating-Profile
